spark2.4/src/main/scala/KafkaReader.scala

```diff
@@ -45,23 +47,33 @@ object KafkaReader {
 //
 //    val metricsProducer = new KafkaProducer[Integer, Long](props)
 //    val metricsTopic = "ssds-metrics"
-
+    
+    // Create schema from ddl string like
+    //    "ID STRING NOT NULL, LOCATION STRING, TEMPERATURE DOUBLE, HUMIDITY DOUBLE, TM TIMESTAMP"
+    var schema = new StructType
+    schemaDDL.split(",").foreach( s => {
```

Comment:
```
foreach { s => 
  ...
}
```

When you start with `{` you can avoid `{` after `=>`.

Reply:
You could also use [StructType.fromDDL](https://github.com/apache/spark/blob/290a528bff7bcb449714c1c6f1885bd0f804358d/sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala#L539).

Reply:
Changed foreach format in commit https://github.com/splicemachine/splice-machine-spark-connector/pull/27/commits/43f72a64c5a3c29c4609801af6e8f428e4fde6aa

I do prefer using functions from a library and tried to use StructType.fromDDL, but it didn’t support "NOT NULL".
`scala> StructType.fromDDL("ID STRING NOT NULL, LOCATION STRING, TEMPERATURE DOUBLE, HUMIDITY DOUBLE, TM TIMESTAMP")`
`org.apache.spark.sql.catalyst.parser.ParseException:
mismatched input 'NOT' expecting <EOF>(line 1, pos 10)
`